src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIResponse.java

```diff
@@ -115,13 +117,28 @@ public boolean isSuccess() {
   }
 
   private void checkRetryable() {
-    Gson gson = new Gson();
-    JsonObject jo = gson.fromJson(this.responseBody, JsonObject.class);
-    if (jo.get(ServiceNowConstants.STATUS) != null &&
-      jo.get(ServiceNowConstants.STATUS).getAsString().equals(ServiceNowConstants.FAILURE) &&
-      jo.getAsJsonObject(ServiceNowConstants.ERROR).get(ServiceNowConstants.MESSAGE).getAsString()
-        .contains(ServiceNowConstants.MAXIMUM_EXECUTION_TIME_EXCEEDED)) {
+    if (httpStatus == HTTP_STATUS_TOO_MANY_REQUESTS) {
       isRetryable = true;
+      this.responseBody =
```

Comment:
If we know the status is 429, why we need to create a json string, then parse it again just to get the `status` field as `failure`? Seem like we are overly complicating things here.